vcsclient/github.go

```diff
@@ -224,7 +224,7 @@ func (client *GitHubClient) DownloadRepository(ctx context.Context, owner, repos
 	}
 	client.logger.Info("extracted repository successfully")
 	return vcsutils.CreateDotGitFolderWithRemote(localPath, "origin",
-		fmt.Sprintf("https://github.com/%s/%s.git", owner, repository))
+		fmt.Sprintf("%s/%s/%s.git", strings.TrimSuffix(client.vcsInfo.APIEndpoint, "/"), owner, repository))
```

Comment:
Since this `fmt.Sprintf` is duplicated in the code, and is used for both GitHub and GitLab, let's move it into a function inside `utils.go` and also add a test for it, to ensure the output URL has the expected structure.

Reply:
done